framework/data/Pagination.php

```diff
@@ -126,7 +126,7 @@ class Pagination extends BaseObject implements Linkable
     /**
      * @var int total number of items.
      */
-    public $totalCount = 0;
+    public $totalCount = -1;
```

Comment:
`-1` is a special value that decreases overall clarity of the class. We can change it to `null` instead but that would require manually initializing `totalCount` for the case where it's `0`.